File: chrome/test/data/webui/md_history/test_util.js
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/*
 * Waits for queued up tasks to finish before proceeding. Inspired by:
 * https://github.com/Polymer/web-component-tester/blob/master/browser/environment/helpers.js#L97
 */
function flush(callback) {
  Polymer.dom.flush();
  window.setTimeout(callback, 0);
}

File: components/autofill_assistant/browser/test_util.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/autofill_assistant/browser/test_util.h"

namespace autofill_assistant {

bool operator==(const google::protobuf::MessageLite& proto_a,
                const google::protobuf::MessageLite& proto_b) {
  std::string serialized_proto_a;
  proto_a.SerializeToString(&serialized_proto_a);

  std::string serialized_proto_b;
  proto_b.SerializeToString(&serialized_proto_b);

  return serialized_proto_a == serialized_proto_b;
}

}  // namespace autofill_assistant
